.cypress/integration/ad/dashboard/ad_dashboard.spec.ts

```diff
@@ -13,31 +13,47 @@
  * permissions and limitations under the License.
  */
 
-/// <reference types="cypress" />
-
-import {
-  AD_PATH,
-  AD_URL,
-  API_URL_PREFIX,
-  APP_URL_PREFIX,
-  DASHBOARD,
-  DETECTORS,
-  SLASH,
-} from '../../../utils/constants';
+import { DASHBOARD } from '../../../utils/constants';
+import { buildAdAppUrl } from '../../../utils/helpers';
 
 context('AD Dashboard', () => {
-  it('Empty dashboard - no detector', () => {
-    cy.server();
-    cy.route(
-      'GET',
-      [API_URL_PREFIX, AD_PATH, DETECTORS + '*'].join(SLASH),
-      'fixture:no_detector_index_response.json'
-    ).as('getDetectors');
+  it('Empty dashboard - no detector index', () => {
+    cy.mockGetDetectorOnAction('no_detector_index_response.json', () => {
+      cy.visit(buildAdAppUrl(DASHBOARD));
+    });
+    cy.contains('h2', 'You have no detectors');
+  });
 
-    cy.visit([APP_URL_PREFIX, AD_URL, DASHBOARD].join(SLASH));
+  it('Empty dashboard - empty detector index', () => {
+    cy.mockGetDetectorOnAction('empty_detector_index_response.json', () => {
+      cy.visit(buildAdAppUrl(DASHBOARD));
+    });
+    cy.contains('h2', 'You have no detectors');
+  });
 
-    cy.wait('@getDetectors', { requestTimeout: 60_000 });
+  it('AD dashboard - single stopped detector', () => {
+    cy.mockGetDetectorOnAction('single_detector_response.json', () => {
+      cy.visit(buildAdAppUrl(DASHBOARD));
+    });
 
-    cy.contains('h2', 'You have no detectors');
+    cy.contains('h3', 'Live anomalies');
+    cy.contains(
+      'p',
+      'All matching detectors are under initialization or stopped for the last 30 minutes. Please adjust filters or come back later.'
+    );
+  });
+
+  it('AD dashboard - redirect to create detector', () => {
+    cy.mockGetDetectorOnAction('no_detector_index_response.json', () => {
+      cy.visit(buildAdAppUrl(DASHBOARD));
+    });
+
+    cy.mockSearchIndexOnAction('search_index_response.json', () => {
+      cy.get('.euiButton--primary.euiButton--fill:first').click({
```

Comment:
Can we use a simpler way to get the button? Like via name?

Reply:
'Create detector' text is inside `<span>`, which is inside the button element `<a>`
![Screen Shot 2020-06-16 at 2 07 20 PM](https://user-images.githubusercontent.com/59710443/84828383-d3d67500-afda-11ea-916d-ad9ecf89172c.png)

Getting it via name only returns the `<span>`, which is not clickable. 

The simpler way is to use `span[data-test-subj="add_detector"]`